docs/adding-new-chain.md

```diff
@@ -0,0 +1,152 @@
+# Adding New Chain Support
+
+> Back to [CONTRIBUTING.md](../CONTRIBUTING.md)
+
+This guide walks through implementing a new chain family in the CCIP SDK.
+
+## Overview
+
+A chain implementation consists of:
+
+1. **SDK Implementation** (`ccip-sdk/src/{chainFamily}/index.ts`) - Lane discovery, token operations, message tracking, transaction building
+1. **Hasher** (`ccip-sdk/src/{chainFamily}/hasher.ts`) - Computes unique message IDs from CCIP message contents
```

Comment:
Not sure if I'd consider hasher split in its own file a core requirement; instead, it's more a natural pattern for a chain family's implementation organization; this is just needed/required for fulfilling `Chain.getDestLeafHasher` static method, which falls into requirement above (i.e. you need to implement this method for `Chain` type check to pass)